spec/models/concerns/hyku_addons/user_behavior_spec.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+RSpec.describe HykuAddons::UserBehavior, type: :model do
+  subject(:user) { create(:user) }
+
+  describe ".with_public_profile" do
+    subject(:query) { User.with_public_profile }
+    let(:public_user) { create(:user, display_profile: true) }
+
+    before do
+      user && public_user
+    end
+
+    it "doesn't return private users" do
+      expect(query).to include(public_user)
+      expect(query).not_to include(user)
+    end
+
+    it "doesn't affect normal queries" do
+      expect(User.all.count).to eq 2
+    end
+  end
+  describe "#save" do
+    let(:creator) do
+      [
+        {
+          creator_family_name: "Hawking",
+          creator_given_name: "Stephen",
+          creator_organization_name: "",
+          creator_institutional_email: user.email,
+          creator_profile_visibility: User::PROFILE_VISIBILITY[:closed]
+        }
+      ].to_json
+    end
+    let(:work) { create(:work, creator: [creator]) }
+
+    it "enqueues the ToggleDisplayProfileJob job when display_profile setting is updated" do
+      user.display_profile = true
+
+      expect { user.save }.to have_enqueued_job(HykuAddons::ToggleDisplayProfileJob)
+    end
+
+    it "does not enque a job unless the display profile setting has been changed" do
+      user.department = "software development"
+
+      expect { user.save }.not_to have_enqueued_job(HykuAddons::ToggleDisplayProfileJob)
+    end
+  end
+
+  describe "#display_profile" do
+    it "defaults to false" do
```

Comment:
I'm presuming this means when you create a user with `display_profile =nil` then it gets set to false.  What happens if you try to update a user and set `display_profile = nil`?

Reply:
@thenapking `display_profile` has a db level default of `false` on a boolean column, so i dont think it can be `nil`.

Reply:
Great!